docs/migrations.md

```diff
@@ -228,3 +231,95 @@ If commit is `True` (default),
 
 `stream_data` must be a list of `dict`-like objects
 containing the blocks within the given StreamField.
+
+## Recreating migrations
+
+[As described above](#schema-migrations), 
+each time a Django model's definition changes it requires the generation of a new Django migration. 
+Over time, the number of migrations in our apps can grow very large, 
+slowing down testing and the `migrate` command.
+
+For this reason it may be desirable to periodically delete and recreate the migration files, 
+so that instead of a series of files detailing every change over time 
+we have a smaller set that just describes the current state of models in the code. 
+
+Django does provide an automated 
+[squashing](https://docs.djangoproject.com/en/1.11/topics/migrations/#squashing-migrations) 
+process for migrations, 
+but this is often not optimal when migrations contain manual `RunPython` blocks that we don't necessarily care about keeping around.
+
+Instead, we delete all existing migration files and then run `manage.py makemigrations` to create new ones. 
+This will generate the smallest number of migration files needed to describe the state of models in the code; 
+typically one per app although sometimes multiple are needed due to app dependencies.
+
+This process does have these critical side effects:
+
+1. Databases that exist at some migration state before the one at the point of the recreation will no longer be able to be migrated to the current state, 
+    as the intermediate changes will have been lost. 
+    This means that those databases will need to be recreated. 
+    This also means that historical database archives will require a bit more work to resurrect; 
+    they'll need to first be migrated to the point just before the recreation, 
+    and then updated to code at or after that point.
+
+    For example, say a database dump exists at a point where N migrations occur. 
+    At such time as N + 1 migrations occur, 
+    we decide to go through the recreation process. 
+    Now we have a new migration numbered N + 2 that represents the equivalent of all (1..N+1) migrations that it replaces. 
+    If you try to load and migrate the dump at point N, 
+    Django no longer has the code necessary to go from N->N+1 only -- 
+    it only has the ability to go from 0->N+2. 
+    To recover such a dump, you'll need to check out the code at the point before the recreation was done, 
+    migrate from N->N+1, and then check out latest and migrate forwards.
+
+2. Any open pull requests at the time of the recreation that reference 
+    or depend on some of the existing migrations 
+    will need to be modified to instead refer to the new migration files.
+
+Migrations can be recreated with this process:
+
+1. Remove all existing migration files:
```

Comment:
One tricky problem: what about other repositories ([teachers-digital-platform](https://github.com/cfpb/teachers-digital-platform/blob/master/teachers_digital_platform/migrations/0001_initial.py#L15)) that depend on cfgov-refresh migrations?

Reply:
I think we resolved this in chat — but as long as the squashed migration has the `includes` list in step 4, it should be fine.